### PR TITLE
feat:issues page

### PR DESCRIPTION
**Pull request by e-roy**
this PR adds basic UI for `/issue` page

issue #14

### CODE REVIEW

File: src/app/issues/page.tsx [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+  const [search, setSearch] = React.useState<string>(
+    `is:open is:issue author:xxxxx archived:false`
+  );
+  const [issues, setIssues] = React.useState<IIssueData[]>(openData);
+  const [issueStatus, setIssueStatus] = React.useState<"open" | "closed">(
+    "open"
+  );
+
+  React.useEffect(() => {
```
Comment (pierresisson) [inline src/app/issues/page.tsx:29]:
```suggestion
  useEffect(() => {

```

File: src/app/issues/page.tsx [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            <span className="isolate inline-flex rounded-md shadow-sm w-full">
+              <button
+                type="button"
+                className={clsx(
+                  "relative inline-flex w-full md:w-min items-center rounded-l-md px-4 py-1.5 font-semibold text-slate-300 ring-1 ring-inset ring-slate-600 focus:z-10",
+                  {
+                    "bg-purple-600 text-slate-50": issueType === `created`,
+                  }
+                )}
+                onClick={() => setIssueType("created")}
```
Comment (pierresisson) [inline src/app/issues/page.tsx:51]: make a callback before the return

File: src/app/issues/page.tsx [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+              </button>
+              <button
+                type="button"
+                className={clsx(
+                  "relative -ml-px inline-flex w-full md:w-min items-center rounded-r-md px-4 py-1.5 font-semibold text-slate-300 ring-1 ring-inset ring-slate-600 focus:z-10",
+                  {
+                    "bg-purple-600 text-slate-50": issueType === `mentioned`,
+                  }
+                )}
+                onClick={() => setIssueType("mentioned")}
```
Comment (pierresisson) [inline src/app/issues/page.tsx:75]: make a callback before the return

File: src/app/issues/page.tsx [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+          <label className="relative w-full border-slate-600 text-slate-400 border rounded-md">
+            <span className="sr-only">Search</span>
+            <span className="absolute inset-y-0 left-0 flex items-center px-2">
+              <Search className="h-4 w-4" />
+            </span>
+            <input
+              className="block bg-[#0E1116] w-full h-full rounded-md py-1 pl-9 pr-3 focus:outline-none focus:border-purple-500 focus:ring-purple-500 focus:ring-1 text-sm md:text-base"
+              type="text"
+              value={search}
+              onChange={(e) => setSearch(e.target.value)}
```
Comment (pierresisson) [inline src/app/issues/page.tsx:91]: make a callback before the return

File: src/app/issues/page.tsx [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+                    <CircleDot className="h-5 w-5 mr-2 mt-0.5" /> 3 Open
+                  </button>
+                  <button
+                    className={clsx(
+                      "flex text-slate-400 hover:text-slate-200",
+                      {
+                        "text-slate-200": issueStatus === `closed`,
+                      }
+                    )}
+                    onClick={() => setIssueStatus("closed")}
```
Comment (pierresisson) [inline src/app/issues/page.tsx:119]: make a callback before the return

File: src/app/issues/page.tsx [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+                    Sort <ChevronDown className="h-4 w-4 ml-1 mt-1.5" />
+                  </span>
+                </div>
+              </div>
+            </div>
+            <div className="overflow-hidden rounded-md rounded-tr-none rounded-tl-none border border-t-0 dark:border-gray">
+              <ul className="divide-y dark:divide-gray">
+                {issues.map((item, index) => (
+                  <li
+                    key={index}
```
Comment (pierresisson) [inline src/app/issues/page.tsx:141]: never put an index as key

**Review by pierresisson — CHANGES_REQUESTED** [review]

File: src/app/issues/page.tsx (hunk 2 above)
Comment (e-roy) [reply src/app/issues/page.tsx:53]: @pierresisson please elaborate or give an example

**Comment by pierresisson** [comment]
@e-roy can't you use the react native hook `useId()`?

**Comment by e-roy** [comment]
@pierresisson the link in my previous comment was a link I found in the older react docs.  Apparently from their twitter account they say it's more of a use case for accessibility then for data id.

[https://twitter.com/reactjs/status/1510334505252491267](https://twitter.com/reactjs/status/1510334505252491267)

sure, I can concatenation and remove the libraries.  I was considering for future once backend data is figured out.

this might also bring up the thought of handling mock data until that point also.

**Comment by PeerRich** [comment]
btw @e-roy added you with write access to repo

**Comment by e-roy** [comment]
@pierresisson I'm still looking for an elaboration or example on `make a callback before the return`

I'm a self taught dev, so any info so I can understand what you mean would be much appreciated 🙏🏻

**Comment by pierresisson** [comment]
@e-roy making a pull request from yours to show you!

**Comment by PeerRich** [comment]
why do i see only this: 

CleanShot 2023-03-07 at 17 34 18@2x

**Comment by e-roy** [comment]
@PeerRich because that hasn't been worked on yet.  This PR works on this page [https://nostr-ladu8lx2o-nostrgit.vercel.app/issues](https://nostr-ladu8lx2o-nostrgit.vercel.app/issues)

**Comment by PeerRich** [comment]
> @PeerRich because that hasn't been worked on yet. This PR works on this page https://nostr-ladu8lx2o-nostrgit.vercel.app/issues

ahh cheers!

File: src/app/issues/page.tsx [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+              className="block bg-[#0E1116] w-full h-full rounded-md py-1 pl-9 pr-3 focus:outline-none focus:border-purple-500 focus:ring-purple-500 focus:ring-1 text-sm md:text-base"
+              type="text"
+              value={search}
+              onChange={(e) => setSearch(e.target.value)}
+            />
+          </label>
+        </div>
+
+        <main>
+          <div className="mt-4 rounded-md border dark:border-[#383B42]">
```
Comment (PeerRich) [inline src/app/issues/page.tsx:98]:
```suggestion
          <div className="mt-4">

```

Comment (PeerRich) [reply src/app/issues/page.tsx:98]: there is a double border otherwise

**Comment by PeerRich** [comment]
btw. if you reopen the branch from within this repo the vercel runs automatically

**Comment by e-roy** [comment]
@pierresisson I think I covered the callbacks you are discussing :)

**Comment by cypherhoodlum** [comment]
@PeerRich I think Vercel might be causing more confusion than convenience as it is right now. I cannot see why it fails when it fails and others probably can't either. Can you?

 
Screenshot from 2023-03-08 12-52-46

**Comment by e-roy** [comment]
to see the vercel error you have to be on the team #111 

I've got a vercel account, but I'm not on the team.

I'm also terrible with git commands so I'm flying extra blind 😅

but I am extra excited to iterate off this page 😃

**Comment by pierresisson** [comment]
@e-roy i don't know what is wrong with your pr but i can't link yours to mine
https://github.com/NostrGit/NostrGit/pull/126

my changes are:
```javascript
const handleIssueType = useCallback(() => {
    if (issueType === "created") {
      setIssueType("created");
    } else if (issueType === "assigned") {
      setIssueType("assigned");
    } else {
      setIssueType("mentioned");
    }
  }, [issueType]);

  const handleIssueStatus = useCallback(() => {
    if (issueStatus === "open") {
      setIssueStatus("open");
    } else {
      setIssueStatus("closed");
    }
  }, [issueStatus]);
```

**Comment by e-roy** [comment]
error-vercel
@pierresisson I'm running `yarn build` locally and getting this error.  I even commented out most of the file and tried and still got this error. 

I think I'm just going to create a new PR and close this.  Let me try that and see if that works.

**Pull request closed without merging**